### PR TITLE
Add pypi github workflow

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR adds github workflow to build and push packages to both test pypi and main pypi during releases and push to master branch.

### CODE REVIEW

**Comment by lsetiawan** [comment]
@carlosgjs you'd need to set up the `secrets.TEST_PYPI_API_TOKEN` and `secrets.PYPI_API_TOKEN` for this to work.

File: .github/workflows/pypi.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        python -m build
+        echo ""
+        echo "Generated files:"
+        ls -lh dist/
+    - uses: actions/upload-artifact@v3
+      with:
+        name: releases
+        path: dist
+
+  test-built-dist:
```
Comment (carlosgjs) [inline .github/workflows/pypi.yaml:42]: Is it overkill to upload to the test index and the download it back down? It adds complexity and maintenance cost (e.g. needing a separate token). Can we test the local wheel instead? E.g.
```
pip install dist/noisepy_seis-*.whl
python -c "import noisepy.seis; print(noisepy.seis.__version__)"
...
```
That way we are testing the contents of the package (what we are building) and not the PyPi service.

Comment (lsetiawan) [reply .github/workflows/pypi.yaml:42]: It's usually good to test the publishing and ensuring that the package works against the test pypi. This is to ensure that the behavior will work when pushing to the "production" pypi. Additionally this can provide a way to distribute the package with internal users for testing before officially releasing it to the public. This is really the purpose of the test pypi.

Comment (lsetiawan) [reply .github/workflows/pypi.yaml:42]: In regards of the simple wheel test is to ensure that it works since when you put into conda, it will perform those exact checks. Better to do that during testing then after the package is published :)

**Comment by lsetiawan** [comment]
This is a duplicate of #80, but have some differences. Sorry @carlosgjs! Didn't mean to step on your work, just wanted to share my experience with packaging and distribution, preparing for both pypi and conda.

File: .github/workflows/pypi.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  build-artifact:
+    name: Build noisepy-seis package
+    runs-on: ubuntu-20.04
+    if: github.repository == 'mdenolle/NoisePy'
```
Comment (carlosgjs) [inline .github/workflows/pypi.yaml:18]: Why is this needed?

**Review by carlosgjs — APPROVED** [review]

File: .github/workflows/pypi.yaml (hunk 2 above)
Comment (lsetiawan) [reply .github/workflows/pypi.yaml:18]: I do this check, to make sure that this only runs on this repo and not anyone else's fork. This again is probably an overkill since pypi doesn't allow duplicate packages.

**Comment by lsetiawan** [comment]
Closing in favor of #80
